Makefile

```diff
@@ -1,4 +1,4 @@
-CC=gcc
+CC?=gcc
```

Comment:
How about just invoking `make` with the explicit `CC=$CC` argument? Then we don't need to touch Makefiles.